components/Contact/ContactForm.tsx

```diff
@@ -0,0 +1,75 @@
+'use client';
+import React from 'react';
+import { useRouter } from 'next/navigation';
+import { ToastContainer, toast } from 'react-toastify';
+
+import 'react-toastify/dist/ReactToastify.css';
+
+import { sendEmail } from '@/lib/utils';
+import { useTheme } from '@/context/ThemeProvider';
+
+import ToastMessage from './ToastMessage';
+
+const ContactForm = () => {
+  const { mode } = useTheme();
+  const router = useRouter();
+  return (
+    <form
+      action={async (formData) => {
+        const { data, error } = await sendEmail(formData);
+        if (error) {
+          toast.error(
+            'Error sending message, please try again later'
+          );
+        }
+        if (data) {
+          toast.success(
+            ({ closeToast }) => <ToastMessage formData={formData} />,
+            {
+              theme: mode,
+            }
+          );
+        }
+        formData = new FormData();
+        router.refresh();
```

Comment:
formData isn't the state of the form- just the data coming in. So we can't reset the form by clearing the `formData` variable. 

If we want control of the form- we could use the onSubmit instead of the action.


```js
    <form
      onSubmit={(e) => {
        e.preventDefault();
        // grab the form itself
        const form = e.target as HTMLFormElement;
        // grab the data
        const formData = new FormData(form);
         // TODO:
        // send the email, handle errors, show toasts
        
        // clear the form
        form.reset();
      }}
      
```